### PR TITLE
Deprecate ajax in favor of fetch in ember-data

### PR DESCRIPTION
**Pull request by snewcomer**
[Rendered](https://github.com/snewcomer/rfcs/blob/rfc-0782/text/0782-deprecate-ember-data-ajax.md)

close [#782](https://github.com/emberjs/rfcs/issues/782)

### CODE REVIEW

File: text/0782-deprecate-ember-data-ajax.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+With Ember 4.0 dropping jQuery, we should take a hard stance on dropping the use of `ajax` in the next major release. 
+
+## Detailed design
+
+The first step is putting in place a deprecation if an adapter uses `ajax` to make a request. Since `ember-data` switched the default behaviour to use fetch in 4.0, this deprecation will only apply to those who have overrided the default with `useFetch = true` in their adapters.
+
+Second, we will expose fetch related methods on the adapters that are equivalent in use and flexibility as the current `ajax` options.  Note the [`minimum-adapter-interface`](https://github.com/emberjs/data/blob/master/packages/store/addon/-private/ts-interfaces/minimum-adapter-interface.ts) does not assert the method by which a request is made. As a result, how we design and implement these methods should not have an effect on those users who have implemented their own adapter. 
+
+In 5.0, all `ajax` methods and supporting infrastructure will be removed.
```
Comment (snewcomer) [inline text/0782-deprecate-ember-data-ajax.md:30]: add - list of methods deleted, renamed, added

File: text/0782-deprecate-ember-data-ajax.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## How we teach this
+
+This is best taught through a deprecation guide. Users using `ajax` will need to install `ember-fetch` or rely on the default global `fetch` API provided by the browser.  This has some implications, for example, with how query params are serialized.  Users will have to be careful to ensure the behaviour of their API changes with any changes that result from switching to fetch.
+
+## Drawbacks
+
+Some user's technology stacks may be entirely dependent on `ajax` for requesting API data.  This isn't necessarily an easy switch and may require multiple improvements to various layers to use `fetch`. For those users, we can document how they can still implement their own adapter to use `ajax`.
+
```
Comment (snewcomer) [inline text/0782-deprecate-ember-data-ajax.md:39]: If you are depending on ajax, override fetch...apply to users dependent on this.

1. ajaxOptions and ajax prefilter
2. fetch method can use jquery.ajax in order to take advantage of Ajax setup and Ajax prefilter

**Review by runspired — APPROVED** [review]

**Comment by wagenet** [comment]
@snewcomer @runspired any reason we shouldn't merge this?

**Comment by runspired** [comment]
@snewcomer the file name here needs fixed, otherwise this is good to go
